signac/contrib/schema.py

```diff
@@ -135,14 +137,14 @@ def __repr__(self):
         return "{}(<len={}>)".format(type(self).__name__, len(self))
 
     def __contains__(self, key_or_keys):
-        if isinstance(key_or_keys, str):
-            key_or_keys = key_or_keys.split('.')
-        return tuple(key_or_keys) in self._schema
+        if isinstance(key_or_keys, tuple) or isinstance(key_or_keys, list):
+            key_or_keys = '.'.join(key_or_keys)
+        return key_or_keys in self._schema
 
     def __getitem__(self, key_or_keys):
-        if isinstance(key_or_keys, str):
-            key_or_keys = key_or_keys.split('.')
-        return self._schema.__getitem__(tuple(key_or_keys))
+        if isinstance(key_or_keys, tuple) or isinstance(key_or_keys, list):
```

Comment:
Supporting NumPy arrays here is sort of an odd use-case since these keys must be strings, so you'd have to construct a NumPy array of strings. I'm fine allowing that if you'd like since it's good to ducktype in general, but in that case we should definitely use the try/except approach. NumPy arrays are not Sequences, so the `isinstance` check will fail.

Reply:
Yes, use `try`, `except TypeError` instead of checking `isinstance(key_or_keys, Sequence)`.

Reply:
Just make sure to properly profile this, because that might present a significant performance bottleneck.

Reply:
I tried `try`, `except TypeError` but it failed for tuples as it raises `KeyError`.

Reply:
@vishav1771 I'm not sure if I understand. `'.'.join(('a', 'b'))` successfully joins a tuple for me. You can replace it with `except (TypeError, KeyError):` if I'm missing a different case.

Reply:
I tried the following
```
 try :
   return schema.__getitem__(key)
except TypeError: 
   return schema.__getitem__('.'.join(key))
```
Because I'm expecting key to be a string. and if I use `".".join("abc")` it will return `a.b.c`

Reply:
@vishav1771 Ah! The `KeyError` is raised because `a.b.c` doesn't exist. That's a different problem due to strings being iterable, and we don't want that error to be raised/caught. You have to handle the string case separately, there isn't really a better way that I know of.
```python
if isinstance(key, str):
    return schema[key]
else:
    ...
```

Reply:
Okay, Thanks 👍

Reply:
@vishav1771 I looked at your code again and I think it should be fine if tests are passing. You already did what I had suggested.

Reply:
@csadorf I'm not aware of any performance-critical uses of the `ProjectSchema` class, so if you know of someplace where it's heavily used, please let us know. I don't think this has any reason to perform much differently than the previous code.